ofrak_core/tests/components/test_lz4_component.py

```diff
@@ -0,0 +1,275 @@
+Test the functionality of the LZ4 component, including unpacking,
+modifying, and repacking LZ4-compressed data.
+
+Requirements Mapping:
+- REQ1.3
+- REQ4.4
+"""
+import lz4.frame
+from pathlib import Path
+from typing import Tuple
+
+import pytest
+
+from ofrak.ofrak_context import OFRAKContext
+from ofrak.resource import Resource
+from ofrak.core.lz4 import Lz4ModernData, Lz4SkippableData
+from pytest_ofrak.patterns.compressed_filesystem_unpack_modify_pack import (
+    CompressedFileUnpackModifyPackPattern,
+)
+
+ASSETS_DIR = Path(__file__).parent / "assets"
+
+
+@pytest.fixture(
+    autouse=True,
+    scope="module",
+    params=[
+        (ASSETS_DIR / "hello_world", ASSETS_DIR / "hello_ofrak"),
+    ],
+    ids=["hello world"],
+)
+def lz4_test_input(request):
+    initial_path, repacked_path = request.param
+    with open(initial_path, "rb") as initial_file:
+        initial_data = initial_file.read()
+    with open(repacked_path, "rb") as repacked_file:
+        expected_repacked_data = repacked_file.read()
+    return (initial_data, expected_repacked_data)
+
+
+class Lz4UnpackModifyPackPattern(CompressedFileUnpackModifyPackPattern):
+    """
+    Template for tests that test different inputs the LZ4 component should support
+    unpacking.
+
+    This test verifies that:
+    - An LZ4 file can be successfully unpacked
+    - Modifications to the unpacked data can be applied
+    - The modified data can be repacked back into a valid LZ4 file
+    - The LZ4 file contains the expected data after decompression
+    """
+
+    expected_tag = Lz4ModernData
+
+    @pytest.fixture(autouse=True)
+    def create_test_file(self, lz4_test_input: Tuple[bytes, bytes], tmp_path: Path):
+        self.INITIAL_DATA, self.EXPECTED_REPACKED_DATA = lz4_test_input
+        lz4_path = tmp_path / "test.lz4"
+        self.write_lz4(lz4_path)
+        self._test_file = lz4_path.resolve()
+
+    def write_lz4(self, lz4_path: Path):
+        """
+        Write LZ4 compressed data to file.
+
+        :param lz4_path: Path to write the LZ4 file
+        """
+        raise NotImplementedError()
+
+    async def verify(self, repacked_root_resource: Resource):
+        """
+        Verify that the repacked LZ4 resource contains the expected data.
+
+        :param repacked_root_resource: The repacked resource to verify
+        """
+        patched_decompressed_data = lz4.frame.decompress(await repacked_root_resource.get_data())
+        assert patched_decompressed_data == self.EXPECTED_REPACKED_DATA
+
+
+class TestLz4UnpackModifyPack(Lz4UnpackModifyPackPattern):
+    """
+    Test the basic unpack, modify, and pack functionality for a simple LZ4 file.
+
+    This test verifies that:
+    - A standard LZ4 file (modern frame format) can be unpacked
+    - The unpacked data can be modified
+    - The modified data can be repacked into a valid LZ4 file
+    """
+
+    def write_lz4(self, lz4_path: Path):
+        compressed_data = lz4.frame.compress(self.INITIAL_DATA)
```

Comment:
We probably want to have static test files instead of doing compression on the fly. That way we're not testing the LZ4 library against itself, but rather testing it against real LZ4 instances found in the wild.